text/0007-gas-price-adjustment.md

```diff
@@ -0,0 +1,135 @@
+# Daily Gas Price Adjustment Algorithm
+
+## Summary
+
+[summary]: #summary
+
+CEP PR: [casperlabs/ceps#0004](https://github.com/casperlabs/ceps/pull/0004)
+
+Gas price is regulated algorithmically. A price floor exists and adjusts daily, based on fullnesses of blocks from the previous day, in order to absorb volatility and reduce congestion.
+
+## Motivation
+
+[motivation]: #motivation
+
+Gas prices are known to be subject to high volatility both in the short term and the long term. This is due to a combination of factors such as inelastic supply, [daily demand cycle](https://solmaz.io/2019/10/21/gas-price-fee-volatility/) and shifting trends.
+
+During times of high demand for a blockchain, one can observe prices change up to 10-100x in a matter of hours. A floating gas price and the absence of a reference point exacerbates the situation. Such volatility makes it difficult for dapp providers and users to plan ahead their expenses.
```

Comment:
`Such volatility makes it difficult for dapp providers and users to budget for their expenses`